pkg/controller/registry/resolver/resolver.go

```diff
@@ -27,12 +28,14 @@ type OperatorResolver interface {
 type SatResolver struct {
 	cache cache.OperatorCacheProvider
```

Comment:
Just a preliminary review: given this is a new type of constraint, we should have a test case for it in resolver_test.go to ensure the compound constraint will get resolved into what we expect it to be.

Reply:
I'll be pushing up a bunch of tests today